Unable to install AntlrWorks-Plugin for NetBeans 8.0.2

Hello

First I would thank you guys at TunnelVision for the great work.

Unfotunaltely the plugin seems to be incomaptible with NetBeans 8.0.2. Trying to install it, I'm receiving the following error message:

Some plugins require plugin Editor Settings Storage to be installed.
The plugin Editor Settings Storage is requested in implementation version 201408251540.  The following plugin is affected:       ANTLRWorks Editor 
Some plugins require plugin Editor Options to be installed.
The plugin Editor Options is requested in implementation version 201408251540.  The following plugin is affected:       ANTLRWorks Editor

My NetBeans installation:

Product Version: NetBeans IDE 8.0.2 (Build 201411181905)
Updates: NetBeans IDE is updated to version NetBeans 8.0.2 Patch 2
Java: 1.8.0_31; Java HotSpot(TM) 64-Bit Server VM 25.31-b07
Runtime: Java(TM) SE Runtime Environment 1.8.0_31-b13
System: Linux version 3.16.7-21-desktop running on amd64; UTF-8; de_DE (nb)

It would be nice, if you'll find the time to look into this.

Thank you very much!


Best regards

Alex

Hi again. I guess I'm probably missing something.
Having correctly set the enviroment variables as you pointed above, but for Mac OS X:
```
ant clean
Buildfile: /Volumes/EMPTY/antlrworks/sam/build.xml

download:

BUILD FAILED
/Volumes/EMPTY/antlrworks/sam/build.xml:7: The following error occurred while executing this line:
/Volumes/EMPTY/antlrworks/sam/nbproject/build-impl.xml:42: Cannot find NetBeans build harness.
Check that nbplatform.nb801.netbeans.dest.dir and nbplatform.nb801.harness.dir are defined.
On a developer machine these are normally defined in ${user.properties.file}=${netbeans.user}/build.properties
but for automated builds you should pass these properties to Ant explicitly.
You may instead download the harness and platform: -Dbootstrap.url=.../tasks.jar -Dautoupdate.catalog.url=.../updates.xml

Total time: 0 seconds
```

So, I deduced I need to provide the platform and harness directories. I used my current (stable) NetBeans installations for this purpose:
```
export suite_dir='/Applications/NetBeans/NetBeans 8.0.1.app/Contents/Resources/NetBeans'
export netbeans_dest_dir=${suite_dir}/platform
export harness_dir=${suite_dir}/harness
```

Then, executed:
```
ant -Dnbplatform.nb801.netbeans.dest.dir="$netbeans_dest_dir" -Dnbplatform.nb801.harness.dir="$harness_dir" clean
```
which finished successfully:
```
...
clean:

BUILD SUCCESSFUL
Total time: 1 second
```

However, after trying to build:
```
ant -Dnbplatform.nb801.netbeans.dest.dir="$netbeans_dest_dir" -Dnbplatform.nb801.harness.dir="$harness_dir" build

...
build-init:
    [mkdir] Created dir: /Volumes/EMPTY/antlrworks/sam/build/public-package-jars

BUILD FAILED
/Applications/NetBeans/NetBeans 8.0.1.app/Contents/Resources/NetBeans/harness/suite.xml:187: The following error occurred while executing this line:
/Applications/NetBeans/NetBeans 8.0.1.app/Contents/Resources/NetBeans/harness/build.xml:174: java.io.IOException: Invalid ${cluster.path.final}:
	at org.netbeans.nbbuild.ModuleListParser.<init>(ModuleListParser.java:779)
	at org.netbeans.nbbuild.ParseProjectXml.execute(ParseProjectXml.java:451)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:441)
	at org.apache.tools.ant.taskdefs.SubAnt.execute(SubAnt.java:306)
	at org.apache.tools.ant.taskdefs.SubAnt.execute(SubAnt.java:221)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	at org.apache.tools.ant.Main.runBuild(Main.java:851)
	at org.apache.tools.ant.Main.startAnt(Main.java:235)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)

Total time: 0 seconds
```

What kind of NetBeans platform configuration do you use?

Hello Sam

Sorry, it took me some time...But at last I've some results...
Unfortunately, these results aren't looking this good...

While building, this error was thrown:

Generating ANTLR 4 grammars:
  org/antlr/works/editor/grammar/experimental/GrammarLexer.g4
  org/antlr/works/editor/grammar/experimental/GrammarParser.g4
  org/antlr/works/editor/grammar/highlighter/GrammarHighlighterLexer.g4
  org/antlr/works/editor/st4/experimental/TemplateLexer.g4
  org/antlr/works/editor/st4/experimental/TemplateParser.g4
  org/antlr/works/editor/st4/highlighter/GroupHighlighterLexer.g4
error(7):  cannot find or open file: *.g4
error(7):  cannot find or open file: *.g4
error(7):  cannot find or open file: *.g4
error(7):  cannot find or open file: *.g4
/home/Alexander/Spielwiese/ntebans_harness/suite.xml:187: The following
error occurred while executing this line:
/home/Alexander/Spielwiese/ANTLRWorksPlugin/antlrworks2-bfee49a9b029625ef3bffeaa1b026d2bbc95b61d/org-antlr-works-editor/build.xml:120:
The following error occurred while executing this line:
/home/Alexander/Spielwiese/ANTLRWorksPlugin/antlrworks2-bfee49a9b029625ef3bffeaa1b026d2bbc95b61d/org-antlr-works-editor/build.xml:95:
Java returned: 1
BUILD FAILED (total time: 24 seconds)

I'll try to figure this out...

Best regards

Alex

Am 13.06.2015 um 17:36 schrieb Sam Harwell:
> Hi Alex,
> 
> It's no problem. What operating system do you have? I'm not the best at
> explaining how to use Git, but I'm sure I could find someone who can help.
> 
> If you don't want to use Git, you can simply download a Zip file
> containing all the source from the repository. This will let you build
> the project, but won't let you send your changes back to us for review.
> It's the highlighted button here:
> 
> image
> <https://cloud.githubusercontent.com/assets/1408396/8144911/f18c0202-11b7-11e5-890b-05cbbc67aca8.png>
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tunnelvisionlabs/antlrworks2/issues/49#issuecomment-111720947>.
>

--- a/.gitignore
+++ b/.gitignore
@@ -5,6 +5,7 @@
 
 # ignore download cache
 */lib/cache/
+/lib/
 
 # ignore build directories
 build/

--- a/build.xml
+++ b/build.xml
@@ -5,4 +5,71 @@
 <project name="ANTLRWorks2" basedir=".">
     <description>Builds the module suite ANTLRWorks2.</description>
     <import file="nbproject/build-impl.xml"/>
+
+    <target name="init-ant-contrib">
+        <property name="ant-contrib.groupId" value="ant-contrib"/>
+        <property name="ant-contrib.artifactId" value="ant-contrib"/>
+        <property name="ant-contrib.version" value="1.0b3"/>
+        <property name="ant-contrib.file" value="lib/${ant-contrib.artifactId}.jar"/>
+    </target>
+
+    <target name="clean" depends="suite.clean,init-ant-contrib">
+        <delete includeemptydirs="true" failonerror="false" file="${ant-contrib.file}"/>
+    </target>
+
+    <target name="download-ant-contrib" depends="init-ant-contrib">
+        <property name="suite.cache.dir" value="lib/cache"/>
+        <property name="ant-contrib.intermediatefile" value="${suite.cache.dir}/${ant-contrib.artifactId}-${ant-contrib.version}.jar"/>
+
+        <mkdir dir="${suite.cache.dir}"/>
+        <get src="https://repo1.maven.org/maven2/${ant-contrib.groupId}/${ant-contrib.artifactId}/${ant-contrib.version}/${ant-contrib.artifactId}-${ant-contrib.version}.jar"
+             dest="${ant-contrib.intermediatefile}"
+             skipexisting="true"/>
+        <copy file="${ant-contrib.intermediatefile}" tofile="${ant-contrib.file}"/>
+    </target>
+
+    <target name="nbms" depends="init-keystore,suite.nbms">
+    </target>
+
+    <target name="init-keystore" depends="-init">
+        <!-- Create/update keystore -->
+        <delete file="${keystore.location}${keystore.name}"/>
+        <mkdir dir="${keystore.location}"/>
+        <genkey alias="${keystore.alias}" storepass="${keystore.password}"
+            dname="${keystore.dname}"
+            keystore="${keystore.location}${keystore.name}"/>
+
+        <!-- Update keystore info in projects -->
+        <antcall target="update-keystore-info"/>
+    </target>
+
+    <target name="update-keystore-info" depends="download-ant-contrib">
+        <taskdef resource="net/sf/antcontrib/antlib.xml">
+            <classpath>
+                <pathelement location="${ant-contrib.file}"/>
+            </classpath>
+        </taskdef>
+
+        <for list="${modules}" delimiter=":" param="cur" trim="true">
+            <sequential>
+                <!-- Place the keystore information in the properties file -->
+                <mkdir dir="@{cur}/nbproject/"/>
+                <propertyfile file="@{cur}/nbproject/project.properties">
+                    <entry key="keystore" value="../${keystore.location}${keystore.name}"/>
+                    <entry key="nbm_alias" value="${keystore.alias}"/>
+                </propertyfile>
+                <replaceregexp file="@{cur}/nbproject/project.properties" match="^#.*\n" replace=""/>
+
+                <!-- Place the password in the private properties file -->
+                <mkdir dir="@{cur}/nbproject/private/"/>
+                <propertyfile file="@{cur}/nbproject/private/platform-private.properties">
+                    <entry key="storepass" value="${keystore.password}"/>
+                </propertyfile>
+            </sequential>
+        </for>
+
+        <propertyfile file="nbproject/private/platform-private.properties">
+            <entry key="storepass" value="${keystore.password}"/>
+        </propertyfile>
+    </target>
 </project>

--- a/editor.completionext/nbproject/project.properties
+++ b/editor.completionext/nbproject/project.properties
@@ -1,5 +1,5 @@
 is.autoload=true
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/editor.containingfolder/nbproject/project.properties
+++ b/editor.containingfolder/nbproject/project.properties
@@ -1,7 +1,7 @@
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 license.file=../Eula.html
-nbm.homepage=http://tunnelvisionlabs.com
+nbm.homepage=http\://tunnelvisionlabs.com
 nbm.module.author=Sam Harwell
 project.license=tvl-prop
 keystore=../nbproject/private/keystore

--- a/nbproject/platform.properties
+++ b/nbproject/platform.properties
@@ -9,4 +9,4 @@ cluster.path=\
     ${nbplatform.active.dir}/profiler:\
     ${nbplatform.active.dir}/websvccommon
 nbjdk.active=JDK_1.7__x64_
-nbplatform.active=nb801
+nbplatform.active=nb802

--- a/nbproject/project.properties
+++ b/nbproject/project.properties
@@ -35,5 +35,10 @@ project.org.tvl.editor.containingfolder=editor.containingfolder
 project.org.tvl.modules.editor.completion=editor.completionext
 project.org.tvl.netbeans.editor.actions=tvl-editor-actions
 project.org.tvl.netbeans.editor.whitespace=tvl.editor.whitespace
+keystore.dname=CN=Sam Harwell, OU=NetBeans Development, O=Tunnel Vision Laboratories\\, LLC, L=Helotes, ST=TX, C=US
+keystore.location=nbproject/private/
+keystore.name=keystore
+keystore.alias=myself
+keystore.password=specialsauce
 keystore=nbproject/private/keystore
 nbm_alias=myself

--- a/org-abego-treelayout-core/nbproject/project.properties
+++ b/org-abego-treelayout-core/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
-javac.compilerargs=-Xlint -Xlint:-serial -bootclasspath ${bootclasspath.java6}
+javac.compilerargs=-Xlint -Xlint\:-serial -bootclasspath ${bootclasspath.java6}
 javac.source=1.6
 license.file=../LICENSE-abego.txt
-nbm.homepage=http://code.google.com/p/treelayout/
+nbm.homepage=http\://code.google.com/p/treelayout/
 nbm.module.author=abego Software GmbH
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/org-antlr-netbeans/nbproject/project.properties
+++ b/org-antlr-netbeans/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 license.file=../Eula.html
-nbm.homepage=http://tunnelvisionlabs.com
+nbm.homepage=http\://tunnelvisionlabs.com
 nbm.module.author=Sam Harwell
 project.license=tvl-prop
 keystore=../nbproject/private/keystore

--- a/org-antlr-runtime/nbproject/project.properties
+++ b/org-antlr-runtime/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
-javac.compilerargs=-Xlint -Xlint:-serial -bootclasspath ${bootclasspath.java6}
+javac.compilerargs=-Xlint -Xlint\:-serial -bootclasspath ${bootclasspath.java6}
 javac.source=1.6
 license.file=../LICENSE-ANTLR.txt
-nbm.homepage=http://www.antlr.org/
+nbm.homepage=http\://www.antlr.org/
 nbm.module.author=Terence Parr
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/org-antlr-v4-annotations/nbproject/project.properties
+++ b/org-antlr-v4-annotations/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
-javac.compilerargs=-Xlint -Xlint:-serial -bootclasspath ${bootclasspath.java6}
+javac.compilerargs=-Xlint -Xlint\:-serial -bootclasspath ${bootclasspath.java6}
 javac.source=1.6
 license.file=../LICENSE-ANTLR.txt
-nbm.homepage=http://www.antlr.org/
+nbm.homepage=http\://www.antlr.org/
 nbm.module.author=Terence Parr, Sam Harwell
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/org-antlr-v4-runtime/nbproject/project.properties
+++ b/org-antlr-v4-runtime/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
-javac.compilerargs=-Xlint -Xlint:-serial -bootclasspath ${bootclasspath.java6}
+javac.compilerargs=-Xlint -Xlint\:-serial -bootclasspath ${bootclasspath.java6}
 javac.source=1.6
 license.file=../LICENSE-ANTLR.txt
-nbm.homepage=http://www.antlr.org/
+nbm.homepage=http\://www.antlr.org/
 nbm.module.author=Terence Parr, Sam Harwell
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/org-antlr-v4/nbproject/project.properties
+++ b/org-antlr-v4/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
-javac.compilerargs=-Xlint -Xlint:-serial -bootclasspath ${bootclasspath.java6}
+javac.compilerargs=-Xlint -Xlint\:-serial -bootclasspath ${bootclasspath.java6}
 javac.source=1.6
 license.file=../LICENSE-ANTLR.txt
-nbm.homepage=http://www.antlr.org/
+nbm.homepage=http\://www.antlr.org/
 nbm.module.author=Terence Parr, Sam Harwell
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/org-antlr-works-editor/nbproject/project.properties
+++ b/org-antlr-works-editor/nbproject/project.properties
@@ -1,7 +1,7 @@
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 license.file=../Eula.html
-nbm.homepage=http://tunnelvisionlabs.com
+nbm.homepage=http\://tunnelvisionlabs.com
 nbm.module.author=Sam Harwell
 project.license=tvl-prop
 keystore=../nbproject/private/keystore

--- a/org-antlr/nbproject/project.properties
+++ b/org-antlr/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
-javac.compilerargs=-Xlint -Xlint:-serial -bootclasspath ${bootclasspath.java6}
+javac.compilerargs=-Xlint -Xlint\:-serial -bootclasspath ${bootclasspath.java6}
 javac.source=1.6
 license.file=../LICENSE-ANTLR.txt
-nbm.homepage=http://www.antlr.org/
+nbm.homepage=http\://www.antlr.org/
 nbm.module.author=Terence Parr
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/org-stringtemplate-v4/nbproject/project.properties
+++ b/org-stringtemplate-v4/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
-javac.compilerargs=-Xlint -Xlint:-serial -bootclasspath ${bootclasspath.java6}
+javac.compilerargs=-Xlint -Xlint\:-serial -bootclasspath ${bootclasspath.java6}
 javac.source=1.6
 license.file=../LICENSE-ANTLR.txt
-nbm.homepage=http://www.stringtemplate.org/
+nbm.homepage=http\://www.stringtemplate.org/
 nbm.module.author=Terence Parr
 keystore=../nbproject/private/keystore
 nbm_alias=myself

--- a/tvl-editor-actions/nbproject/project.properties
+++ b/tvl-editor-actions/nbproject/project.properties
@@ -1,7 +1,7 @@
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 license.file=../Eula.html
-nbm.homepage=http://tunnelvisionlabs.com
+nbm.homepage=http\://tunnelvisionlabs.com
 nbm.module.author=Sam Harwell
 nbm.needs.restart=true
 project.license=tvl-prop

--- a/tvl.editor.whitespace/nbproject/project.properties
+++ b/tvl.editor.whitespace/nbproject/project.properties
@@ -1,7 +1,7 @@
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 license.file=../Eula.html
-nbm.homepage=http://tunnelvisionlabs.com
+nbm.homepage=http\://tunnelvisionlabs.com
 nbm.module.author=Sam Harwell
 project.license=tvl-prop
 keystore=../nbproject/private/keystore

--- a/works.editor.antlr3/nbproject/project.properties
+++ b/works.editor.antlr3/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 license.file=../Eula.html
-nbm.homepage=http://tunnelvisionlabs.com
+nbm.homepage=http\://tunnelvisionlabs.com
 nbm.module.author=Sam Harwell
 project.license=tvl-prop
 keystore=../nbproject/private/keystore

--- a/works.editor.antlr4/nbproject/project.properties
+++ b/works.editor.antlr4/nbproject/project.properties
@@ -1,8 +1,8 @@
 is.autoload=true
 javac.source=1.7
-javac.compilerargs=-Xlint -Xlint:-serial
+javac.compilerargs=-Xlint -Xlint\:-serial
 license.file=../Eula.html
-nbm.homepage=http://tunnelvisionlabs.com
+nbm.homepage=http\://tunnelvisionlabs.com
 nbm.module.author=Sam Harwell
 project.license=tvl-prop
 keystore=../nbproject/private/keystore

